### DISCUSSION TITLE
How to get a value from a joined table

### DISCUSSION THREAD

**Discussion by s-ohnishi** [phalcon/cphalcon#15688] [Q&A]
I made the following model.
```PHP
namespace My\Models;
use Phalcon\Mvc\Model

class Users extends Model
{
    public int $id;
    public string $name;
    public int $bar;
}

class Books extends Model
{
    public int $id;
    public int $usersId;
    public string $name;
    public int $foo;
}
```
And I am writing like this 
```PHP
$uRows = \My\Models\Users::query()
    ->Columns(['My\Models\Users.*', 'bid'=>'B.id']
    ->leftJoin('My\Models\Bools', 'My\Models\Users.id=B.userId', 'B')
    ->execute();

foreach($uRows as $usr) {
    echo $usr->readAttribute('my\Models\Users')->id, ':::', $usr->bid, PHP_EOL;
}
```
in order to get this result.
```SQL
SELECT users.*, B.id AS bid FROM users LEFT JOIN books AS B ON users.is=B.users_id;
```

It's important to note that readAttribute () should specify a string with a lowercase letter at the beginning of the original namespace.

It feels very confusing to always have to pass the correct namespace-name through lcfirst() before calling readAttribute().
If the model is placed in the global space, it makes sense that the table name and class name are automatically converted, but if it is not, it is difficult to do it without permission.
And I think that everything should not be placed in the global space, but should be placed in a clearly separated namespace.
"My\Models\Users" and "my\Models\Users" probably won't exist at the same time, but I think they have the problem of making namespaces indistinguishable.

If you specify aliases for all in Columns(), readAttribute() becomes unnecessary, but it is very troublesome if there are many columns.
If you can specify a table alias like Join(), you can call it with the specified alias, so you don't have to worry about case, but I think there is no way.

How can I use it with minimal effort and without mistakes?
Is there any way other than writing it carefully?

**Comment by quasipickle** [comment]
I've never actually seen that `query()` method used.  Whenever someone uses the query builder they usually just use the regular query builder class.  Doing so will allow you to set an alias when you specify the table to pull from.

_assuming this code is in a controller, or some other space that has access to `modelsManager`_

```php
$uRows = $this->modelsManager->createBuilder()
    ->addFrom(\My\Models\Users::class, 'user')
    ->Columns(['user.*', 'bid'=>'B.id']
    ->leftJoin('My\Models\Books', 'user.id=B.userId', 'B')
    ->execute();

foreach($uRows as $row) {
    echo $row->user->id,':::',$row->bid, PHP_EOL;
}
```

**Reply by s-ohnishi** [reply]
> I've never actually seen that query() method used.

Congratulations on your first experience.

> Whenever someone uses the query builder they usually just use the regular query builder class. 

Why not use it?
Is it because the function (alias setting in this case) is insufficient?
It may be better not to implement methods that are not used by anyone due to lack of functionality.
But I'm pretty sure which model is the main one, so I'd like to see an alias feature (I don't know if there's anything else missing) added to `query()`.


I was misunderstanding.
Namespace names are not case sensitive in PHP, so my\Models\Users and My\Models\Users should be equated.
https://www.php.net/manual/en/language.namespaces.rationale.php
The matter I mentioned in the question cannot be a problem.

However, I feel that `readAttribute('my\Models\Users')` can be acquired well, and `readAttribute('My\Models\Users')` cannot be acquired according to the PHP specifications.
I think both should work in the same way, but what about?
